### PR TITLE
bouton FranceConnect

### PR DESCRIPTION
**Pull request by pknoth**
connects to https://github.com/betagouv/api-particulier/issues/109

### CODE REVIEW

**Comment by pknoth** [comment]
connects to betagouv/api-particulier#109

File: components/form.js [hunk 1]
```diff
@@ -51,6 +55,13 @@ class Form extends React.Component {
         this.getSiren()
       })
     }
+
+    setTimeout(() => {
```
Comment (rdubigny) [inline components/form.js]: I am sure there is a better solution than this setTimeout. I'll be glad to help you on this :)

File: package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "jsdom": "^11.6.2",
     "lodash": "^4.17.10",
     "next": "^5.1.0",
+    "lodash": "^4.17.10",
```
Comment (rdubigny) [inline package.json]: lodash is already set at line 29. Please remove this line

File: components/form.js [hunk 3]
```diff
@@ -240,10 +272,14 @@ class Form extends React.Component {
...
-          <label htmlFor='intitule_demarche'>Intitulé</label>
-          <input type='text' onChange={this.handleChange} name='demarche.intitule' id='intitule_demarche' disabled={readOnly} value={demarche.intitule} />
-        </div>
+        { !form.franceConnected && (
+          <div className='form__group'>
+            <label htmlFor='intitule_demarche'>Intitulé</label>
+            <input type='text' onChange={this.handleChange} name='demarche.intitule' id='intitule_demarche' disabled={readOnly} value={demarche.intitule} />
+          </div>
+        )
+        }
```
Comment (rdubigny) [inline components/form.js]: fancy linting: put the closing parenthesis and the closing bracket on the same line

File: components/form.js [hunk 4]
```diff
@@ -164,12 +176,16 @@ class Form extends React.Component {
         validation_de_convention,
         id
       },
+      serviceProviders,
       sirenNotFound,
       errors
     } = this.state
+    const france_connect_authorize_uri = FRANCE_CONNECT_AUTHORIZE_URI
```
Comment (rdubigny) [inline components/form.js]: 1. what is the point of this ?
2. we should never create snakecased variable in the front

File: components/form.js [hunk 5]
```diff
@@ -199,6 +215,22 @@ class Form extends React.Component {
...
+        { form.franceConnected && (
+          <div className='form__group'>
+            <h2 id='france-connect'>Partenaire FranceConnect</h2>
+            <p><Link href={france_connect_authorize_uri}><a className='button'>Se connecter auprès de France Connect afin de récupérer mes démarches</a></Link></p>
+            <label htmlFor='fournisseur_de_service'>Intitulé de la démarche</label>
+            <select onChange={this.handleChange} name='enrollment.fournisseur_de_service'>
+              <option>-- sélectionnez une démarche --</option>
+              {
+                serviceProviders.map(serviceProvider => {
+                  return <option key={i++} selected={fournisseur_de_service === serviceProvider.name} value={serviceProvider.name}>{serviceProvider.name}</option>
```
Comment (rdubigny) [inline components/form.js]: are you sure there is no attribute in service provider that we can use as a key ? Also there is no point of incrementing an index manually it results in the same number of rendering as using the map index.

File: components/form.js [hunk 6]
```diff
@@ -199,6 +215,22 @@ class Form extends React.Component {
         <h1>{form.text.title}</h1>
         <div dangerouslySetInnerHTML={{__html: form.text.intro}} className='intro' />
 
+        { form.franceConnected && (
+          <div className='form__group'>
+            <h2 id='france-connect'>Partenaire FranceConnect</h2>
+            <p><Link href={france_connect_authorize_uri}><a className='button'>Se connecter auprès de France Connect afin de récupérer mes démarches</a></Link></p>
+            <label htmlFor='fournisseur_de_service'>Intitulé de la démarche</label>
+            <select onChange={this.handleChange} name='enrollment.fournisseur_de_service'>
+              <option>-- sélectionnez une démarche --</option>
```
Comment (rdubigny) [inline components/form.js]: this option is more a placeholder. You should at least disable it (see https://stackoverflow.com/a/42316092/2590861)

File: lib/user.js [hunk 7]
```diff
@@ -50,6 +50,9 @@ class User {
 
   getServiceProviders() {
     const token = localStorage.getItem('token-fc')
+    if (!token) {
+      return Promise.reject()
```
Comment (rdubigny) [inline lib/user.js]: please add a reason in the rejection

**Pull request merged**